components/react/src/App.js

```diff
@@ -1,56 +1,52 @@
-import React, { Component } from 'react';
+import React, { useEffect, useState } from 'react';
 import Header from './Components/Header';
 import ProjectList from './Components/ProjectList';
 import SelectorMediator from './Components/SelectorMediator';
 
-class App extends Component {
-  abortController = new AbortController();
-
-  constructor(props) {
-    super(props);
-    this.state = {
-      isLoaded: false,
-      projects: [],
-    };
-  }
-
-  componentDidMount() {
-    this.request('http://localhost:5455/projects.json', 'projects')
-      .then(() => this.request('http://localhost:5455/taxonomies.json', 'taxonomies'))
-      .then(() => this.setState(Object.assign({}, this.state, { isLoaded: true })));
-  }
-
-  request(url, stateVariable) {
-    return fetch(url, { signal: this.abortController.signal })
+const App = () => {
+  const abortController = new AbortController();
+  const [isLoaded, setIsLoaded] = useState(false);
+  const [projects, setProjects] = useState([]);
+  const [taxonomies, setTaxonomies] = useState([]);
+  const [error, setError] = useState();
+  const [selected, setSelected] = useState({});
+  const [matchScores, setMatchScores] = useState({});
+
+  useEffect(
+    () => request('projects', setProjects).then(() => request('taxonomies', setTaxonomies)).then(() => setIsLoaded(true)),
+    [],
+  );
+  const request = (endpoint, setter) => {
+    return fetch(`http://localhost:5455/${endpoint}.json`, { signal: abortController.signal })
```

Comment:
According to React, we should use `async/await` instead.

Reply:
Agreed, this should be wrapped in a `Promise`. We can do that using `async/await` or just create a `Promise` object.

Reply:
Fixed, now this uses `async/await` to make the fetch.